packages/plugin-core/src/workspace.ts

```diff
@@ -659,21 +661,155 @@ export class DendronWorkspace {
+        vscode.ViewColumn.One,
+        {
+          enableScripts: true,
+        }
+      );
+      panel.webview.html = content;
+
+      panel.webview.onDidReceiveMessage(
+        async (message) => {
+          switch (message.command) {
+            case "initializeWorkspace":
+              await new SetupWorkspaceCommand().run({
+                workspaceInitializer: new TutorialInitializer(),
+              });
+              return;
+          }
+        },
+        undefined,
+        undefined
+      );
     } catch (err) {
       vscode.window.showErrorMessage(JSON.stringify(err));
     }
   }
 }
+
+/**
+ * Type that can execute custom code as part of workspace creation and opening of a workspace.
+ */
+export type WorkspaceInitializer = {
+  /**
+   * Invoked after workspace has been created. Perform operations such as copying over notes.
+   */
+  onWorkspaceCreation?: (opts: { vaults: DVault[]; wsRoot: string }) => void;
+
+  /**
+   * Invoked after the workspace has been opened. Perform any operations such as re-arranging the layout.
+   */
+  onWorkspaceOpen?: (opts: { ws: DendronWorkspace }) => void;
+};
+
+/**
+ * Factory class for creating WorkspaceInitializer types
+ */
+export class WorkspaceInitFactory {
+  static create(ws: DendronWorkspace): WorkspaceInitializer | undefined {
+    if (this.isTutorialWorkspaceLaunch(ws.context)) {
+      return new TutorialInitializer();
+    }
+
+    return;
+  }
+
+  private static isTutorialWorkspaceLaunch(
+    context: vscode.ExtensionContext
+  ): boolean {
+    const state = context.globalState.get<string | undefined>(
+      GLOBAL_STATE.WORKSPACE_ACTIVATION_CONTEXT
+    );
+    return (
+      _.isUndefined(state) ||
+      state === WORKSPACE_ACTIVATION_CONTEXT.TUTORIAL.toString()
+    );
+  }
+}
+
+/**
+ * Workspace Initializer for the Tutorial Experience. Copies tutorial notes and
+ * launches the user into the tutorial layout after the workspace is opened.
+ */
+export class TutorialInitializer implements WorkspaceInitializer {
+  onWorkspaceCreation = async (opts: { vaults: DVault[]; wsRoot: string }) => {
+    const ws = DendronWorkspace.instance();
+
+    await ws.updateGlobalState(
+      GLOBAL_STATE.WORKSPACE_ACTIVATION_CONTEXT,
+      WORKSPACE_ACTIVATION_CONTEXT.TUTORIAL.toString()
+    );
+
+    const dendronWSTemplate = VSCodeUtils.joinPath(
+      ws.extensionAssetsDir,
+      "dendron-ws"
+    );
+
+    const vpath = vault2Path({ vault: opts.vaults[0], wsRoot: opts.wsRoot });
+
+    fs.copySync(path.join(dendronWSTemplate.fsPath, "tutorial"), vpath);
+
+    // Tailor the tutorial text to the particular OS and for their workspace location.
+    const options = {
+      files: [path.join(vpath, "*.md")],
+
+      from: [/%KEYBINDING%/g, /%WORKSPACE_ROOT%/g],
+      to: [
+        process.platform === "darwin" ? "Cmd" : "Ctrl",
+        path.join(opts.wsRoot, "dendron.code-workspace"),
+      ],
+    };
+
+    const replace = require("replace-in-file");
```

Comment:
instead of `require` can you use `import`? this way, typechecking will kick in if it has types. if not, you can add a custom type declaration instead